squirrel.worker/src/test/java/org/dice_research/squirrel/analyzer/impl/MimeTypeDetectorTest.java

```diff
@@ -35,16 +42,32 @@ public void initialize() {
         classLoader = getClass().getClassLoader();
     }
 
-    public Lang validate(String fileName) {
-        File file = new File(classLoader.getResource(fileName).getFile());
-        return typeDetector.detectMimeType(file);
+    private Lang validate(String fileName) {
+        if(fileName == null) {
+            return RDFLanguages.RDFNULL;
+        }
+        else {
+            URL url = classLoader.getResource(fileName);
+            if(url != null) {
+                File file = new File(url.getFile());
+                return typeDetector.detectMimeType(file);
+            }
+            else {
+                return RDFLanguages.RDFNULL;
+            }
+        }
     }
 
     @Parameterized.Parameters
     public static Collection filesToTest() {
         return Arrays.asList(new Object[][]{
-            {"rdf_analyzer/new_york/new_york_rdf", RDFLanguages.RDFXML},
-            {"sample.ttl", RDFLanguages.TURTLE}
+            {"rdf_analyzer/new_york/RDFXML", RDFLanguages.RDFXML},
+            {"Sample_Files/Test_File_2", RDFLanguages.RDFNULL},
+            {"rdf_analyzer/new_york/new_york_turtle", RDFLanguages.TURTLE},
+            {"Sample_Files/sample.nt", RDFLanguages.NTRIPLES},
+            {"rdf_analyzer/new_york/new_york_rdfjson", RDFLanguages.RDFJSON},
+            {"rdf_analyzer/new_york/new_york_jsonld", RDFLanguages.JSONLD},
+            {null, RDFLanguages.RDFNULL}
```

Comment:
From what I see, the files for turtle, RDFJSON and JSON-LD are missing.

However, as explained below, using the New York resource as example does not make much sense since they will be very large.

Btw. the last test case does not make much sense, since you are handling it inside your own test class. So you are not really testing the mime type detector but your test class.

Reply:
I have created sample files for all the automata's to check and also removed the null check which was not sensible.